crates/builder/src/config.rs

```diff
@@ -101,7 +106,7 @@ impl Commands {
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, Serialize, Deserialize)]
 pub struct Config {
```

Comment:
The `metadata` field can be removed and replaced with `target_directory: Utf8PathBuf` as that is the only field that is used by the rest of the code. Like that there will be much less data serialized to file.
The `builder_deps` could be removed, no?